lib/makeConstructor.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+// core modules
+var util = require('util');
+
+// external modules
+var _ = require('lodash');
+var assert = require('assert-plus');
+
+// local files
+var RestError = require('./baseClasses/RestError');
+
+
+
+/**
+ * create RestError subclasses for users. takes a string, creates a
+ * constructor for them. magicks, again.
+ * @public
+ * @function makeConstructor
+ * @param {String} name the name of the error class to create
+ * @param {Number} defaults optional status code
+ * @return {Function} a constructor function
+ */
+function makeConstructor(name, defaults) {
+
+    assert.string(name, 'name');
+    assert.optionalObject(defaults, 'defaults');
+
+    // code property doesn't have 'Error' in it. remove it.
+    var defaultCode = name.replace(new RegExp('[Ee]rror$'), '');
+    var d = _.assign({}, {
```

Comment:
I know it's not in the linter but I prefer avoiding one letter variables as it's hard to know what they mean and also hard to search and replace them.

Reply:
Good call, will rename